oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBeanImpl.java

```diff
@@ -122,13 +133,13 @@ public TabularData getIndexStats() throws IOException {
             TabularType tt = new TabularType(LuceneIndexMBeanImpl.class.getName(),
                     "Lucene Index Stats", IndexStats.TYPE, new String[]{"path"});
             tds = new TabularDataSupport(tt);
-            Set<String> indexes = indexTracker.getIndexNodePaths();
-            for (String path : indexes) {
+            // Use indexPathService to get list of all the lucene indexes.
```

Comment:
done

Reply:
@nit0906 I think this change is sound from a logical perspective. I am wondering though if this can cause some performance issues. The `indexPathService` returns all the indexes, even the ones that are not used (eg: an old version of an index). When we call `indexTracker.acquireIndexNode` (line 142) the `IndexSearcher` gets initialized. If the index is not present it gets downloaded. We can potentially download a lot of indexes only to serve the stats.

Reply:
@fabriziofortino  - Agreed with your point about the performance. 

We can ignore older versions of indexes here (using the same filter logic as is used in the FullTextIndex while creating the index plans) - but I am wondering if that would be correct ? Because for statistics, we would want to show the older indexes as well - if they are active (being indexed) and/or consuming disk space. 

Thoughts ? (cc : @thomasmueller  @tihom88 )

Reply:
@fabriziofortino  - how about, I send the isActive info in the IndexStats constructor and not call the getSearcher() for old/inactive indexes. 
This way, we can get size for them (which might be relevant). And we don't calculate the numDocs etc (for which we open the searcher).

Reply:
It sounds good to me.

Reply:
@fabriziofortino  - it seems we init the searcher when we do acuireIndexNode from IndexTracker - so we wouldn't be able to avoid that. 

So it's either get stats for all indexes (disabled/old/active). Or get stats only for active ones and ignore disabled and old indexes completely.  We can't do the half and half approach as I thought earlier.

Reply:
We need to avoid downloading all indexes, that's for sure. If that means we have some unknown data (numDocs and so on), then that's fine. Should we set "unknown" to the value -1 or 0?

Reply:
I hope some of the info we can still list, even without potentially downloading the indexes (by calling acquireIndexNode) - for example lastUpdatedTimestamp, hasHiddenOakLibsMount, hasPropertyIndex, and size on disk. But I guess we can't list things like nrtNumDocs, numDocs, and so on. Right?

Reply:
@thomasmueller  - We can get lastUpdatedTimestamp, hasHiddenOakLibsMount, hasPropertyIndex, but not size on disk, as currently that is also calculated using IndexTracker's get Readers.